src/core_gpu_tests/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+###############################################################################
+### CORE GPU UNIT TESTS ###
+
+#
+# The GPU unit tests are based on the black box approach which means that
+# the framework is exclusively using the OpenColorIO dynanic library.
+#
+
+add_definitions("-DOCIO_SOURCE_DIR=${CMAKE_SOURCE_DIR}")
+
+include_directories(
+    ${CMAKE_SOURCE_DIR}/src/core_gpu_tests
+    ${CMAKE_SOURCE_DIR}/export/
+    ${CMAKE_BINARY_DIR}/export/
+    ${OPENGL_INCLUDE_DIR}
+    ${GLUT_INCLUDE_DIR}
+    ${GLEW_INCLUDES}
+    )
+
+set(EXTERNAL_COMPILE_FLAGS "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
+
+if(WIN32)
+    # Hack to get around the "Debug" and "Release" sub directories 
+    #  cmake tries to add with Windows VC+++ IDE
+
+    # Set output path
+    set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} )
+    set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} )
+    set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} )
+    # For multi-config builds (e.g. msvc)
+    foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
+        string( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )
+        set( CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${CMAKE_CURRENT_BINARY_DIR} )
+        set( CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${CMAKE_CURRENT_BINARY_DIR} )
+        set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${CMAKE_CURRENT_BINARY_DIR} )
+    endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
+
+    # Mute a design issue where the Exception public class inherits 
+    # from a STL Exception. STL classes are never supposed to
+    # be exported among different dynamic libraries.
+    set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} /wd4275")
```

Comment:
This should be updated after the OCIO Exception is modified

Reply:
Unfortunately, there are two issues around STL classes in public API: 1) the `std::string` and 2) the `std::exception`. The previous PR removed the `std::string` (easy change as it's a private member, and code is fully backward compatible) but did nothing for the `std::exception` (now a `std::runtime_error`). The later change is much more problematic because some existing code could stop working properly (if the inheritance is removed).

So these lines are still needed to mute some warnings.